lib/temporal/connection/grpc.rb

```diff
@@ -117,9 +118,13 @@ def start_workflow_execution(
 
         client.start_workflow_execution(request)
       rescue ::GRPC::AlreadyExists => e
-        # Feel like there should be cleaner way to do this...
-        run_id = e.details[/RunId: ([a-f0-9]+-[a-f0-9]+-[a-f0-9]+-[a-f0-9]+-[a-f0-9]+)/, 1]
-        raise Temporal::WorkflowExecutionAlreadyStartedFailure.new(e.details, run_id)
+        cast_error = e.to_rpc_status&.details&.map do |any_error|
```

Comment:
nit: `find` should work much better here and should remove the need for `map` / `compact` / `first`

Reply:
Ended up removing `first` so that all error details are available